source/server/server.cc

```diff
@@ -302,8 +302,30 @@ void InstanceImpl::updateServerStats() {
       stats_store_.symbolTable().getRecentLookups([](absl::string_view, uint64_t) {}));
 }
 
+CertStats InstanceImpl::generateCertStats(Stats::Scope& scope) {
+  return {ALL_CERT_STATS(POOL_GAUGE(scope))};
+}
+
+void InstanceImpl::updateCertStats() {
+  auto cert_name_seconds_until_expiration_structs =
+      sslContextManager().getSecondsUntilExpirationOfCerts();
+  for (const auto& cert_name_seconds_until_expiration_struct :
+       cert_name_seconds_until_expiration_structs) {
+    const std::string& cert_name = cert_name_seconds_until_expiration_struct.name;
+    uint64_t expiration_date = cert_name_seconds_until_expiration_struct.expiration;
+    if (cert_stats_map_.find(cert_name) == cert_stats_map_.end()) {
```

Comment:
Stats for certs that no longer exist (due to reconfiguration) need to be deleted.

I wonder if this code should move into the tls transport socket instead of being here (even though the existing daysUntil code is here). The daysUntil needed to be in a common place because it had to look at all the certs in the system and find the one with the shortest expiration.

This new stat is per-cert, and so it feels like it should be attached to the config object that holds it. That would automatically handle deletion when config goes away, and the names could be generated under the existing stat prefix for the transport socket. @jmarantz wdyt?

Reply:
@ggreenway that sounds right to me.

Reply:
@PeterL328 please move all the code into the tls transport socket as described.

Reply:
Will make the changes as described above and let you know when ready to review. Thanks for feedback.

Reply:
Moved code into tls transport socket as described.